src/test/java/org/sonatype/gradle/plugins/scan/nexus/iq/NexusIqScanTaskTest.java

```diff
@@ -99,22 +100,34 @@ public void testScan_real() throws Exception {
     verify(iqClientMock).verifyOrCreateApplication(eq(task.getApplicationId()));
     verify(iqClientMock).getProprietaryConfigForApplicationEvaluation(eq(task.getApplicationId()));
     verify(iqClientMock).evaluateApplication(eq(task.getApplicationId()), eq(task.getStage()),
-        nullable(ScanResult.class), any(File.class));
+        nullable(ScanResult.class), any(File.class), isNull());
   }
 
   @Test
-  public void testScan_realUnabledToCreateApp() throws Exception {
-    NexusIqScanTask task = buildScanTask(false);
+  public void testScan_realWithResultFilePath() throws Exception {
+    NexusIqScanTask task = buildScanTask(false, "some/path");
+    task.setDependenciesFinder(dependenciesFinderMock);
+    when(iqClientMock.verifyOrCreateApplication(eq(task.getApplicationId()))).thenReturn(true);
+
+    task.scan();
+
+    verify(iqClientMock).evaluateApplication(eq(task.getApplicationId()), eq(task.getStage()),
+            nullable(ScanResult.class), any(File.class), eq(new File("some/path")));
+  }
+
+  @Test
+  public void testScan_realUnableToCreateApp() throws Exception {
+    NexusIqScanTask task = buildScanTask(false, null);
     task.setDependenciesFinder(dependenciesFinderMock);
     when(iqClientMock.verifyOrCreateApplication(eq(task.getApplicationId()))).thenReturn(false);
 
-    assertThatThrownBy(() -> task.scan())
+    assertThatThrownBy(task::scan)
         .isInstanceOf(GradleException.class)
         .hasMessageContaining("Application ID test doesn't exist and couldn't be created or the user user doesn't have"
             + " the 'Application Evaluator' role for that application.");
   }
 
-  private NexusIqScanTask buildScanTask(boolean isSimulated) {
+  private NexusIqScanTask buildScanTask(boolean isSimulated, String resultFilePath) {
```

Comment:
Suggestion. I'm tempted to have a overloaded method which call this. That way you don't have to keep supplying a null parameter in the other tests
e.g. 
```
private NexusIqScanTask buildScanTask(boolean isSimulated) {
  return buildScan(isSimulated, null);
}
```

Reply:
> Suggestion. I'm tempted to have an overloaded method which call this one. That way you don't have to keep supplying a null parameter in the other tests
> e.g.
> 
> ```
> private NexusIqScanTask buildScanTask(boolean isSimulated) {
>   return buildScan(isSimulated, null);
> }
> ```

Done: 6dc411735e35fcab3f5cd099dce3e0c254e0ef3c